kernel/aster-nix/src/process/namespaces/mnt_namespace.rs

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::{
+    fs::{path::MountNode, rootfs::root_mount},
+    prelude::*,
+    process::Process,
+};
+
+pub struct MntNamespace {
+    root: Arc<MountNode>,
+}
+
+impl Default for MntNamespace {
+    fn default() -> Self {
+        Self {
+            root: root_mount().clone(),
+        }
+    }
+}
+
+impl MntNamespace {
+    pub fn new(mount_node: Arc<MountNode>) -> Arc<Self> {
+        Arc::new(Self { root: mount_node })
+    }
+
+    pub fn root(&self) -> &Arc<MountNode> {
+        &self.root
+    }
+
+    pub fn copy_mnt_ns(&self, process: &Arc<Process>) -> Arc<Self> {
```

Comment:
I think the API design here is a bit weird. What is the purpose of the parameter `process`?

Reply:
Because the `struct Process` of a new process will be created in clone, we need to move the cwd and root directory of this `struct Process` to the newly created `mount namespace`. The new `mount namespace` actually references a newly copied `mount node tree`. In fact, we need to move the root and cwd directories of the `struct Process` to the new `mount node tree`.

Reply:
After you updated the comments, I understood the meaning of this parameter.